### PR TITLE
Issue #14219: disable symbolic link test cases of property cache file test for windows

### PR DESCRIPTION
**Pull request by piyush-1234**
part of #14219

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java [hunk 1]
```diff
@@ -143,28 +143,50 @@ public void load() throws IOException {
...
-                    directory = actualDir;
-                }
-                else {
-                    throw new IOException(
-                            "Resolved symbolic link " + directory
-                                    + " is not a directory.");
-                }
-            }
-            Files.createDirectories(directory);
+            updateDirectory(directory);
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java:149]: make it like: `OsSpecificUtil.updateDirectory(directory);`

where OsSpecificUtil is static class  in PropertyCacheFile class.

exclude by:
```
<plugin> 
    <groupId>org.jacoco</groupId>
    <artifactId>jacoco-maven-plugin</artifactId>
    <configuration>
        <excludes>
            <exclude>com/baeldung/**/ExcludedPOJO.$OsSpecificUtilclass</exclude>
```

File: src/main/java/com/puppycrawl/tools/checkstyle/Generated.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * reflection.
+ *
+ * <p>{@code @Target({ElementType.TYPE, ElementType.METHOD, ElementType.CONSTRUCTOR})}
+ * specifies the types of elements to which this annotation can be applied.
+ *
+ * @since 1.0
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.METHOD, ElementType.CONSTRUCTOR})
+public @interface Generated {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/Generated.java:65]: remove this class, it is not required.

**Review by romani — CHANGES_REQUESTED** [review]
good, items to improve:

File: src/main/java/com/puppycrawl/tools/checkstyle/Generated.java (hunk 2 above)
Comment (piyush-1234) [reply src/main/java/com/puppycrawl/tools/checkstyle/Generated.java]: Done @romani

File: src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java (hunk 1 above)
Comment (piyush-1234) [reply src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java]: Done @romani

**Comment by romani** [comment]
ci failure - https://app.circleci.com/pipelines/github/checkstyle/checkstyle/22779/workflows/3a56ffe0-f061-4b50-97fb-327952d37172/jobs/470249
please apply this diff:
```
Checking internal (checkstyle website) links only.
--- .ci-temp/linkcheck-suppressions-sorted.txt	2024-01-08 13:02:39.167622074 +0000
+++ .ci-temp/linkcheck-errors-sorted.txt	2024-01-08 13:02:39.159622043 +0000
@@ -721,6 +721,7 @@
 <a href="apidocs/com/puppycrawl/tools/checkstyle/PropertyCacheFile.ExternalResource.html#%3Cinit%3E(java.lang.String,java.lang.String)">#%3Cinit%3E(java.lang.String,java.lang.String)</a>: doesn't exist.
 <a href="apidocs/com/puppycrawl/tools/checkstyle/PropertyCacheFile.html#%3Cinit%3E(com.puppycrawl.tools.checkstyle.api.Configuration,java.lang.String)">#%3Cinit%3E(com.puppycrawl.tools.checkstyle.api.Configuration,java.lang.String)</a>: doesn't exist.
 <a href="apidocs/com/puppycrawl/tools/checkstyle/PropertyCacheFile.html#%3Cinit%3E(com.puppycrawl.tools.checkstyle.api.Configuration,java.lang.String)">../../PropertyCacheFile.html#%3Cinit%3E(com.puppycrawl.tools.checkstyle.api.Configuration,java.lang.String)</a>: doesn't exist.
+<a href="apidocs/com/puppycrawl/tools/checkstyle/PropertyCacheFile.OsSpecificUtil.html#%3Cinit%3E()">#%3Cinit%3E()</a>: doesn't exist.
 <a href="apidocs/com/puppycrawl/tools/checkstyle/PropertyType.html#%3Cinit%3E(java.lang.String)">#%3Cinit%3E(java.lang.String)</a>: doesn't exist.
 <a href="apidocs/com/puppycrawl/tools/checkstyle/SarifLogger.html#%3Cinit%3E(java.io.OutputStream,com.puppycrawl.tools.checkstyle.AbstractAutomaticBean.OutputStreamOptions)">#%3Cinit%3E(java.io.OutputStream,com.puppycrawl.tools.checkstyle.AbstractAutomaticBean.OutputStreamOptions)</a>: doesn't exist.
 <a href="apidocs/com/puppycrawl/tools/checkstyle/SarifLogger.html#%3Cinit%3E(java.io.OutputStream,com.puppycrawl.tools.checkstyle.AbstractAutomaticBean.OutputStreamOptions)">../SarifLogger.html#%3Cinit%3E(java.io.OutputStream,com.puppycrawl.tools.checkstyle.AbstractAutomaticBean.OutputStreamOptions)</a>: doesn't exist.
@@ -1278,6 +1279,7 @@
 <a href="com/puppycrawl/tools/checkstyle/PropertiesExpander.html#%3Cinit%3E(java.util.Properties)">com/puppycrawl/tools/checkstyle/PropertiesExpander.html#%3Cinit%3E(java.util.Properties)</a>: doesn't exist.
 <a href="com/puppycrawl/tools/checkstyle/PropertyCacheFile.ExternalResource.html#%3Cinit%3E(java.lang.String,java.lang.String)">com/puppycrawl/tools/checkstyle/PropertyCacheFile.ExternalResource.html#%3Cinit%3E(java.lang.String,java.lang.String)</a>: doesn't exist.
 <a href="com/puppycrawl/tools/checkstyle/PropertyCacheFile.html#%3Cinit%3E(com.puppycrawl.tools.checkstyle.api.Configuration,java.lang.String)">com/puppycrawl/tools/checkstyle/PropertyCacheFile.html#%3Cinit%3E(com.puppycrawl.tools.checkstyle.api.Configuration,java.lang.String)</a>: doesn't exist.
+<a href="com/puppycrawl/tools/checkstyle/PropertyCacheFile.OsSpecificUtil.html#%3Cinit%3E()">com/puppycrawl/tools/checkstyle/PropertyCacheFile.OsSpecificUtil.html#%3Cinit%3E()</a>: doesn't exist.
 <a href="com/puppycrawl/tools/checkstyle/PropertyType.html#%3Cinit%3E(java.lang.String)">com/puppycrawl/tools/checkstyle/PropertyType.html#%3Cinit%3E(java.lang.String)</a>: doesn't exist.
 <a href="com/puppycrawl/tools/checkstyle/SarifLogger.html#%3Cinit%3E(java.io.OutputStream,com.puppycrawl.tools.checkstyle.AbstractAutomaticBean.OutputStreamOptions)">com/puppycrawl/tools/checkstyle/SarifLogger.html#%3Cinit%3E(java.io.OutputStream,com.puppycrawl.tools.checkstyle.AbstractAutomaticBean.OutputStreamOptions)</a>: doesn't exist.
 <a href="com/puppycrawl/tools/checkstyle/site/ClassAndPropertiesSettersJavadocScraper.html#%3Cinit%3E()">com/puppycrawl/tools/checkstyle/site/ClassAndPropertiesSettersJavadocScraper.html#%3Cinit%3E()</a>: doesn't exist.
------------ grep of linkcheck.html--END
```
to file https://github.com/checkstyle/checkstyle/blob/master/config/linkcheck-suppressions.txt 
to avoid CI violation

**Comment by romani** [comment]
pitest violation is weird, report is at https://github.com/checkstyle/checkstyle/actions/runs/7447762991?pr=14232
code is covered , but something is wrong with class declaration
image

**Comment by piyush-1234** [comment]
> pitest violation is weird, report is at https://github.com/checkstyle/checkstyle/actions/runs/7447762991?pr=14232 code is covered , but something is wrong with class declaration image

checking @romani

**Comment by romani** [comment]
@piyush-1234 , I pushed commit to fix pitest and links, but there is problem now with jacoco, please investigate and fix

**Comment by romani** [comment]
Please fix typo https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=18685&view=logs&j=6a833393-c551-5000-0916-5000fd3b86e7&t=bdbbd129-77f0-5d73-ce7d-db58d34c8358&l=50

**Comment by piyush-1234** [comment]
> I pushed changes to fix all.

yeah, looks like pitest are failing for class OsSpecificUtil

**Comment by romani** [comment]
I merging this quicker as  more and more contributors still experiencing this problem

**Pull request merged**